mesh/mesh.go

```diff
@@ -79,10 +104,23 @@ func (m *Mesh) SetLatestLayer(idx uint32) {
 	}
 }
 
+func (m *Mesh) CreateLayer(lid LayerID) (*Layer, error){
+	layer, err := m.getLayer(lid)//m.orphanBlocks[lid]
+	if err != nil {
+		panic("no blocks received for layer" + strconv.Itoa(int(lid)))
+	}
+
+	m.Log.Info("Ive created a new layer %v with %v blocks %v txs" , lid, len(layer.blocks))
+	l := NewExistingLayer(lid, layer.blocks)
+
+	m.AddLayer(l)
+	return l, nil
+}
+
 func (m *Mesh) AddLayer(layer *Layer) error {
 	m.lMutex.Lock()
 	defer m.lMutex.Unlock()
-	count := LayerID(m.LocalLayer())
+	count := LayerID(m.LatestSeenLayer())
```

Comment:
i'd change the name to something that indicates that we already received
like latestSynced/latestReceived or something